indico/modules/events/controllers/api.py

```diff
@@ -29,11 +30,32 @@ def _process(self, event):
         return EventDetailsForSeriesManagementSchema().jsonify(event)
 
 
-class RHEventCheckEmail(RHEventBase):
-    """Check a person's email when editing a person link."""
+class RHEventCheckEmail(RHProtected):
+    """Check a person's email when editing a person link.
+
+    Access to this endpoint is restricted to those users who can manage the person link's
+    parent object i.e. event, session block, abstract & {sub}contribution.
+    """
+
+    def _process_args(self):
+        self.object_type, self.event, self.object = get_object_from_args()
+        if self.object is None:
+            raise NotFound
+
+    def _check_access(self):
+        if self.object_type == 'event' and not self.object.can_manage(session.user):
+            raise Forbidden
+        if self.object_type == 'block' and not self.object.session.can_manage(session.user):
+            raise Forbidden
+        if self.object_type == 'abstract' and not self.object.can_manage(session.user):
+            raise Forbidden
+        if self.object_type == 'contribution' and not self.object.can_edit(session.user):
+            raise Forbidden
+        if self.object_type == 'subcontribution' and not self.object.can_edit(session.user):
+            raise Forbidden
```

Comment:
```python
obj = self.object.session if self.object_type == 'block' else self.object
if not obj.can_manage(session.user):
    raise Forbidden
```

Reply:
I also have to make the distinction between `can_edit` and `can_manage` so something like this:
```python
obj = self.object.session if self.object_type == 'session_block' else self.object
method = 'can_edit' if self.object_type in ('contribution', 'subcontribution') else 'can_manage'
if not getattr(obj, method)(session.user):
    raise Forbidden
```
At this point I kind of prefer the original dumb code :smile:

Reply:
OK yeah, not great, i missed the different methods. I guess adding `can_search_persons` methods on those objects is overkill? OTOH, this could also be useful for #5024 which probably needs a similar check...

alternatively if you want to keep the individual checks, at least write them in a way so it also fails when none of the object types match - because right now a typic there would skip the access check altogether. for example, `return` in all the success cases, and if none match you raise Forbidden.